### PR TITLE
🏃 Refactor loadbalancer package

### PR DESCRIPTION
**Pull request by hidekazuna**
**What this PR does / why we need it**:

This PR removes floating IP related function and uses networking service instead in loadbalancer package.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hidekazuna** [comment]
recheck

File: pkg/cloud/services/loadbalancer/loadbalancer.go [hunk 1]
```diff
@@ -484,47 +471,6 @@ func checkIfLbExists(client *gophercloud.ServiceClient, name string) (*loadbalan
 	return &lbList[0], nil
 }
 
-func getOrCreateFloatingIP(client *gophercloud.ServiceClient, openStackCluster *infrav1.OpenStackCluster, ip string) (*floatingips.FloatingIP, error) {
-	var fp *floatingips.FloatingIP
-	var err error
-	if ip != "" {
-		fp, err = checkIfFloatingIPExists(client, ip)
```
Comment (jichenjc) [inline pkg/cloud/services/loadbalancer/loadbalancer.go:491]: this logic is removed, where we cover this logic after the refactor?

Comment (hidekazuna) [reply pkg/cloud/services/loadbalancer/loadbalancer.go:491]: This logic was duplicated.
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/networking/floatingip.go#L32

Comment (jichenjc) [reply pkg/cloud/services/loadbalancer/loadbalancer.go:491]: ok, got it

**Pull request merged**